### PR TITLE
Kernel debug: procfs solution

### PR DESCRIPTION
**Pull request by virkha**
Kernel debug via procfs

### CODE REVIEW

File: 05_debug/procfs/Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ifneq ($(KERNELRELEASE),)
+# kbuild part of makefile
+obj-m := debug_module.o
+else
+# normal makefile
+KDIR ?= /lib/modules/`uname -r`/build
```
Comment (ekovalyov) [inline 05_debug/procfs/Makefile:6]: uname -r  gives version of kernel on host computer.
It produces wrong results for cross-compilation.

File: 05_debug/procfs/debug_module.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	}
+	pr_info("passed data: %s\n", buffer);
+
+	while (buffer[i] != 0) {
+		if (buffer[i] >= 'a' && buffer[i] <= 'z') {
+			buffer[i] = buffer[i] - 32;
+			num_char_converted++;
+		}
+		++i;
+	}
```
Comment (ekovalyov) [inline 05_debug/procfs/debug_module.c:117]: What will happen is size of input string exceed PAGE_SIZE ?

File: 05_debug/procfs/Makefile (hunk 1 above)
Comment (virkha) [reply 05_debug/procfs/Makefile:6]: I set KDIR to specific kernel before run make.
Do you mean set absolute path in KDIR var?

File: 05_debug/procfs/debug_module.c (hunk 2 above)
Comment (virkha) [reply 05_debug/procfs/debug_module.c:117]: Page fault will happen and then SIGSEGV -> core dump (.
Sorry, I don't catch what exactly do you mean?

**Review by virkha — COMMENTED** [review]
commented issues

**Review by ekovalyov — APPROVED** [review]

**Pull request merged**